### DISCUSSION TITLE
ScrollArea replace display: table

### DISCUSSION THREAD

**Discussion by Toshinaki** [mantinedev/mantine#7030] [Feature requests]
I've noticed that [ScrollArea](https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/core/src/components/ScrollArea/ScrollAreaViewport/ScrollAreaViewport.tsx) is no longer depends on [Radix Scroll Area](https://www.radix-ui.com/themes/docs/components/scroll-area).

The `display: table` has been a pain to me, since it make the content takes the full width of the children, not the width of the parent. So it's always very difficult to use when I want only vertical scroll.

There has been a fix for this, that
> Replace display: table with flex styles on the parent that allow the viewport to fill container height

https://github.com/radix-ui/primitives/pull/2945
Though the fix is not released yet.

Since mantine ScrollArea no longer depend on radix, maybe we can adapt this fix?

**Comment by gl3nn** [comment]
Yeah in our application we also constantly have to overwrite `display: table` with !important to get desired results.

**Comment by rtivital** [comment]
This change might be added in v8 release as it can be breaking for some cases

**Reply by Toshinaki** [reply]
Nice to hear that!
Do we have a rough roadmap?

**Reply by rtivital** [reply]
No, there is no roadmap

**Comment by borstessi** [comment]
i got this pain now, is there any workaround?

**Reply by rtivital** [reply]
Override these styles with `!important`.

**Reply by jean-gvh** [reply]
Where do I override these styles ? In the global css file ? Im struggling to find the solution.

**Reply by rtivital** [reply]
Inspect element in the dev tools, copy selector

**Reply by jean-gvh** [reply]
Okay, this is what I tried to do. I selected the m_c0783ff9 mantine-ScrollArea-viewport (see picture) and tried to override the styles in my index.css : `.m_c0783ff9.mantine-ScrollArea-viewport {
  display: block !important;
  width: 100% !important;
  height: auto !important;
}` However I still have an issue. 
![image_2024-12-13_114651768](https://github.com/user-attachments/assets/fea427af-9e93-4d70-8524-a44a8695fd4f)
The display:table is here.

**Reply by rtivital** [reply]
You've added these styles to viewport element that does not have display:table. You need to add these styles to the child element `.mantine-ScrollArea-viewport > div` instead.

**Reply by jean-gvh** [reply]
Thanks, indeed it is working now :)

**Comment by icflorescu** [comment]
> Override these styles with `!important`

I did the same on a few different occasions :-)
